orchestrator/db/filters/process.py

```diff
@@ -72,22 +77,48 @@ def target_filter(query: SearchQuery, value: str) -> SearchQuery:
         .filter(ProcessSubscriptionTable.workflow_target.in_(targets))
         .subquery()
     )
-    return query.filter(ProcessTable.pid == process_subscriptions.c.pid)
-
-
-PROCESS_FILTER_FUNCTIONS_BY_COLUMN: dict[str, Callable[[SearchQuery, str], SearchQuery]] = {
-    "processId": generic_is_like_filter(ProcessTable.pid),
-    "istask": generic_bool_filter(ProcessTable.is_task),
-    "assignee": generic_values_in_column_filter(ProcessTable.assignee),
-    "status": generic_values_in_column_filter(ProcessTable.last_status),
-    "workflow": generic_is_like_filter(ProcessTable.workflow),
-    "creator": generic_is_like_filter(ProcessTable.created_by),
-    "product": product_filter,
-    "tag": tag_filter,
-    "subscriptions": subscriptions_filter,
-    "subscriptionId": subscription_id_filter,
-    "target": target_filter,
-}
+    return query.filter(ProcessTable.process_id == process_subscriptions.c.pid)
 
 
+def organisation_filter(query: SearchQuery, value: str) -> SearchQuery:
+    try:
+        value_as_uuid = UUID(value)
+    except (ValueError, AttributeError):
+        msg = f"Not a valid organisation, must be a UUID: '{value}'"
+        logger.debug(msg)
+        raise_status(HTTPStatus.BAD_REQUEST, msg)
+
+    process_subscriptions = (
+        db.session.query(ProcessSubscriptionTable)
+        .join(SubscriptionTable)
+        .filter(SubscriptionTable.customer_id == value_as_uuid)
+        .subquery()
+    )
+    return query.filter(ProcessTable.process_id == process_subscriptions.c.pid)
+
+
+BASE_CAMEL = {to_camel(key): generic_is_like_filter(value) for key, value in inspect(ProcessTable).columns.items()}
+BASE_SNAKE = {key: generic_is_like_filter(value) for key, value in inspect(ProcessTable).columns.items()}
+
+PROCESS_FILTER_FUNCTIONS_BY_COLUMN: dict[str, Callable[[SearchQuery, str], SearchQuery]] = (
+    BASE_CAMEL
+    | BASE_SNAKE
+    | {
+        "isTask": generic_bool_filter(ProcessTable.is_task),
+        "assignee": generic_values_in_column_filter(ProcessTable.assignee),
+        "lastStatus": generic_values_in_column_filter(ProcessTable.last_status),
+        "product": product_filter,
+        "productTag": tag_filter,
+        "subscriptions": subscriptions_filter,
+        "subscriptionId": subscription_id_filter,
+        "target": target_filter,
+        "organisation": organisation_filter,
+        "istask": generic_bool_filter(ProcessTable.is_task),  # TODO: remove in 1.3.0
+        "status": generic_values_in_column_filter(ProcessTable.last_status),  # TODO: remove in 1.3.0
+        "last_status": generic_values_in_column_filter(ProcessTable.last_status),  # TODO: remove in 1.3.0
+        "tag": tag_filter,  # TODO: remove in 1.3.0
+        "creator": generic_is_like_filter(ProcessTable.created_by),  # TODO: remove in 1.3.0
+    }
+)
+
```

Comment:
I think the below example of building the PROCESS_FILTER_FUNCTIONS_BY_COLUMN dict is less error-prone. The `generate_process_filters` and `generate_column_filters` functions (naming is hard) are reusable for subscription filter.

Fine to implement in a follow-up issue as well


```python


def generate_column_filters(
    key: str, f: Callable, extra_keys: list | None
) -> Generator[tuple[str, Callable[[SearchQuery, str], SearchQuery]], None, None]:
    yield key, f
    yield to_camel(key), f

    for extra_key in extra_keys or []:
        yield extra_key, f
        yield to_camel(extra_key), f


column_filters = {
    ProcessTable.is_task: generic_bool_filter,
    ProcessTable.last_status: generic_values_in_column_filter,
    # .. etc
}

column_aliases = {
    ProcessTable.is_task: ["istask"],  # TODO remove in 1.3.0
    # ... etc
}


def generate_process_filters():
    for key, column in inspect(ProcessTable).columns.items():
        extra_keys = column_aliases.get(column, None)
        filter_func = column_filters.get(column, generic_is_like_filter)
        yield from generate_column_filters(key, filter_func, extra_keys=extra_keys)


PROCESS_FILTER_FUNCTIONS_BY_COLUMN_2 = dict(generate_process_filters())
```

Reply:
new issue created for it: https://github.com/workfloworchestrator/orchestrator-core/issues/341